src/Database/Expression/FunctionExpression.php

```diff
@@ -171,8 +172,10 @@ public function sql(ValueBinder $generator)
     {
         $parts = [];
         foreach ($this->_conditions as $condition) {
-            if ($condition instanceof ExpressionInterface) {
-                $condition = sprintf('%s', $condition->sql($generator));
+            if ($condition instanceof Query) {
```

Comment:
Maybe it should be QueryInterface?

Reply:
That would unfortunately only cover ORM queries, as the base query class from the `\Cake\Database` namespace doesn't implement `QueryInterface`.

Reply:
Also `QueryInterface` doesn't even have an `sql()` method :upside_down_face: